### PR TITLE
Pack targetPath always use forward slash

### PR DESCRIPTION
**Pull request by zhili1208**
https://github.com/NuGet/Home/issues/4073

### CODE REVIEW

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs [hunk 1]
```diff
@@ -47,7 +47,8 @@ public string TargetPath
             {
                 if (String.Compare(_targetPath, value, StringComparison.OrdinalIgnoreCase) != 0)
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs:48]: super confusing line, this is `!StringComparer.OrdinalIgnoreCase.Equals(a, b)`

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs [hunk 2]
```diff
@@ -47,7 +47,8 @@ public string TargetPath
...
                 {
-                    _targetPath = value;
+                    // Always use forward slash in package.
+                    _targetPath = value.Replace("\\", "/");
```
Comment (joelverhagen) [inline src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs:51]: We should have a unit test for this 

1. assign `TargetPath`
1. get and validate:
    1. `TargetPath`
    1. `TargetFramework`
    1. `EffectivePath`

**Review by joelverhagen — CHANGES_REQUESTED** [review]
⌚️

File: src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs (hunk 2 above)
Comment (rohit21agrawal) [reply src/NuGet.Core/NuGet.Packaging/PackageCreation/Authoring/PhysicalPackageFile.cs:51]: this doesn't seem like it's right. what if the user actually meant "\" to be a part of his targetPath on a unix system?

**Comment by emgarten** [comment]
🔔 @zhili1208 @rohit21agrawal is this still needed for pack? Can someone merge or close this?

**Comment by mishra14** [comment]
🔔  @zhili1208

**Comment by zhili1208** [comment]
close it

**Pull request closed without merging**